rosette/src/BaseSupp.cc

```diff
@@ -742,24 +744,25 @@ DEF("regexpCompare", regexpCompare, 2, 2) {
     CHECK(0, RBLstring, re);
     CHECK(1, RBLstring, str);
 
-    char *re_s, *str_s;
-    regexp* r;
     Ob* rv;
 
-    re_s = (char*)&re->byte(0);
-    str_s = (char*)&str->byte(0);
+    auto str_s = (char*)&str->byte(0);
+
+    try {
+        std::regex r(re->asCstring()); // Compile the regexp
 
-    if ((r = GS_regcomp(re_s)) && GS_regexec(r, str_s)) {
-        int len = r->endp[0] - r->startp[0];
-        char* res = (char*)malloc(len + 1);
-        strncpy(res, r->startp[0], len);
-        res[len] = '\0';
-        rv = RBLstring::create(res);
-        free(res);
+        if (std::regex_match(str_s, r)) {   // See if it matches
+            rv = RBLstring::create(str_s);
+        } else {
+            rv = RBLFALSE;
+        }
+        return rv;
     }
-    else
+    catch(...) {
         rv = RBLFALSE;
-    return rv;
+        return rv;
+    }
+
```

Comment:
Simplifies a bit into
```C++
DEF("regexpCompare", regexpCompare, 2, 2) {
    CHECK(0, RBLstring, re);
    CHECK(1, RBLstring, str);

    Ob *rv = RBLFALSE;

    auto str_s = (char *)&str->byte(0);

    try {
        std::regex r(re->asCstring()); // Compile the regexp

        if (std::regex_match(str_s, r)) {   // See if it matches
            rv = RBLstring::create(str_s);
        }
    } catch (...) {
        // return PRIM_ERROR("bad regular expression");
    }

    return rv;
}
```

Reply:
Which of course, doesn't require the intermediate `rv` at all and will further simplify to
```C++
DEF("regexpCompare", regexpCompare, 2, 2) {
    CHECK(0, RBLstring, re);
    CHECK(1, RBLstring, str);

    auto str_s = (char *)&str->byte(0);

    try {
        std::regex r(re->asCstring()); // Compile the regexp

        if (std::regex_match(str_s, r)) {   // See if it matches
            return RBLstring::create(str_s);
        }
    } catch (...) {
        // return PRIM_ERROR("bad regular expression");
    }

    return RBLFALSE;
}
```

Reply:
Would you rather see `return PRIM_ERROR(...` than returning RBLFALSE like the current code does?

Reply:
I don't know for sure, but it seems more in line with other primitives. I liked your `warning` idea, too; silent failure is all I'm trying to avoid. You can even take the high road and explain:
```C++
    } catch (const std::regex_error &ex) {
        return __PRIM__->runtimeError(__CTXT__, "bad regular expression: %s", ex.what());
    }
```

Reply:
I decided to not call PRIM_ERROR because the previous implementation just returned RBLFALSE. and PRIM_ERROR acts differently.

Done